src/utilities/plugin-utils.ts

```diff
@@ -0,0 +1,56 @@
+import { Activity } from "../types";
+
+export const loadPluginScripts = (activity: Activity) => {
+  // load any plugin scripts, each should call registerPlugin if correctly loaded
+  // TODO: this URL should come from the activity JSON
+  const teacherEditionPluginURL = "https://teacher-edition-tips-plugin.concord.org/version/v3.5.6/plugin.js";
+  const pluginScripts: string[] = [];
+  for (let page = 0; page < activity.pages.length - 1; page++) {
+    if (!activity.pages[page].is_hidden) {
+      for (let embeddableNum = 0; embeddableNum < activity.pages[page].embeddables.length; embeddableNum++) {
+        const embeddable = activity.pages[page].embeddables[embeddableNum].embeddable;
+        if (embeddable.type === "Embeddable::EmbeddablePlugin" && embeddable.plugin?.component_label === "windowShade") {
+          if (pluginScripts.indexOf(teacherEditionPluginURL) === -1) {
+            pluginScripts.push(teacherEditionPluginURL);
+          }
+        }
+      }
+    }
+  }
+
+  pluginScripts.forEach((pluginScript: string) => {
+    // set plugin label
+    (window as any).LARA.Plugins.setNextPluginLabel("#{v3_plugin_label}");
```

Comment:
Is this correct? I think in the LARA code this gets transformed into a unique string.

from app/view/plugins/_show.haml

```
- v3_plugin_label = "plugin#{plugin.approved_script.id}"
- unless plugin_loaded(plugin.url)
  = content_for :external_scripts do
    :javascript
      LARA.Plugins.setNextPluginLabel("#{v3_plugin_label}");
```
In Ruby `"#{var}"` is equaivalent to JS `` `${var}` ``.

I think this might create an error when we want to add a second plugin.

Reply:
Sheesh.  Good catch!  I didn't even notice that.  Certainly won't work once we get to a second plugin type.  Will fix.

Reply:
I made some adjustments here to use a dynamic string for the label, but this will require further modification as we bring in more plugins.